lib/server/routes/buckets.js

```diff
@@ -1396,6 +1398,129 @@ BucketsRouter.prototype.getStorageUsage = function (req, res) {
   });
 };
 
+BucketsRouter.prototype.finishUpload = function (req, res, next) {
+  const { Bucket, BucketEntry, Upload, Shard, Mirror } = this.storage.models;
+
+  const idBucket = req.params.id;
+
+  const { index, shards } = req.body;
+
+  if (!isHexString(index) || index.length !== 64) {
+    return next(new errors.BadRequestError('Invalid index'));
+  }
+
+  if (!shards) {
+    return next(new errors.BadRequestError('Missing shards'));
+  }
+
+  if (!Array.isArray(shards)) {
+    return next(new errors.BadRequestError('Shards is not an array'));
+  }
+
+  shards.forEach(({ uuid, hash }) => {
+    if (!validateUUID(uuid)) {
+      return next(new errors.BadRequestError('Invalid UUID'));
```

Comment:
Be careful, this is a callback, a return in a callback will not stop the execution of the next iterations nor the `finishUpload` function

Reply:
fixed